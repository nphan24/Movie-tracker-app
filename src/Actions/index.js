export const postMovies = movies => ({
  type: 'POST_MOVIES',
  movies
});

export const addUser = user => ({
  type: 'ADD_USER',
  user
});

export const signInAction = user => ({
  type: 'SIGN_IN',
  user
});

export const validSignIn = error => ({
  type: 'VALID_SIGNIN',
  error
});

export const invalidSignIn = error => ({
  type: 'INVALID_SIGNIN',
  error
});

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
});
