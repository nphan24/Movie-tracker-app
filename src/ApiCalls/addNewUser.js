export const addNewUser = async ({ name, email, password }) => {
  try {
    const response = await fetch("/api/users/new", {
      method: "POST",
      body: JSON.stringify({
        name: name,
        email: email,
        password: password
      }),
      headers: {
        "Content-Type": "application/json"
      }
    });
    const data = await response.json();
    if (data.error) {
      throw new Error(data.error);
    } else {
      return data.id;
    }
  } catch (error) {
    throw new Error(`${email} already exists. Please enter a new email`);
  }
};
