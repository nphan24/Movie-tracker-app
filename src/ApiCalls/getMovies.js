import { cleanMovies } from './cleanMovies';
import { fetchData } from './fetchData';

export const getMovies = async () => {
  const movieData = await fetchData();
  const cleanedMovies = cleanMovies(movieData);
  return cleanedMovies;
};
