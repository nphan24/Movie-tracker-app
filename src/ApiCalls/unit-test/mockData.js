export const mockMovieData = {
  results: [
    {
      original_language: 'en',
      original_title: 'Jumanji: Welcome to the Jungle',
      overview:
        "The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
      popularity: 101.350698,
      poster_path: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
      release_date: '2017-12-09',
      title: 'Jumanji: Welcome to the Jungle',
      video: false,
      vote_average: 6.5,
      vote_count: 3115
    }
  ]
};

export const cleanedMovie = [
  {
    date: 'Release Date: 2017-12-09',
    id: 101.350698,
    image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
    overview:
      "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
    title: 'Jumanji: Welcome to the Jungle',
    vote: 'Rating: 6.5'
  }
];

export const movieInfo = {
  movie_id: '1',
  user_id: '24',
  title: 'hello',
  poster_path: '/hello',
  release_date: '04-01-18',
  vote_average: '-5',
  overview: 'Its not real'
};

export const mockUser = {
  id: '5',
  name: 'r',
  email: 'r@r.r',
  password: 'r'
};

export const mockFavorite = {
  date: 'Release Date: 2017-12-09',
  movie_id: 101.350698,
  image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
  overview:
    "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
  title: 'Jumanji: Welcome to the Jungle',
  vote: 'Rating: 6.5'
};

export const mockRemainingFavorite = {
  date: 'Release Date: 2018-10-5',
  movie_id: 10,
  image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
  overview: 'Summary: The tables are turned as four teenagers.',
  title: 'Fantastic Fours',
  vote: 'Rating: 2.5'
};

export const mockFavoritesArray = [
  {
    date: 'Release Date: 2018-10-5',
    movie_id: 10,
    image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
    overview: 'Summary: The tables are turned as four teenagers.',
    title: 'Fantastic Fours',
    vote: 'Rating: 2.5'
  },
  {
    date: 'Release Date: 2017-12-09',
    movie_id: 101.350698,
    image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
    overview:
      "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
    title: 'Jumanji: Welcome to the Jungle',
    vote: 'Rating: 6.5'
  }
];
