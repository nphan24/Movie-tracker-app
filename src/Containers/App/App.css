.App header {
  background-color: #062a78;
  padding: 40px 20px;
  position: relative;
}

.logo {
  font-family: 'Anton', sans-serif;
  color: #f1cc54;
  font-size: 50px;
  letter-spacing: 3px;
  text-decoration: none;
  border: 3px solid #f1cc54;
  padding: 5px 20px;
}

.logged-in,
.logged-out {
  position: absolute;
  top: 20px;
  right: 20px;
  color: #fff;
}

.logged-in a,
.logged-out a {
  color: #fff;
  text-decoration: none;
  font-size: 20px;
}

.login-menu {
  position: absolute;
  background-color: #f1cc54;
  top: 0;
  left: -120px;
  padding: 20px;
  z-index: 20;
}

.login-menu h2 {
  font-size: 22px;
  margin-bottom: 10px;
}

.login-menu input {
  margin: 5px 0;
  width: 97%;
  font-size: 18px;
  padding: 5px;
  border: none;
  border-radius: 5px;
}

.login-menu button {
  display: block;
  margin-top: 10px;
  font-size: 18px;
  padding: 5px 10px;
  border-radius: 5px;
  border: none;
}

.login-menu .toggle-signin-btn {
  display: block;
  margin-top: 20px;
  text-decoration: underline;
  font-size: 18px;
}
