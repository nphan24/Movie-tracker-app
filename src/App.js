import React, { Component } from 'react';
import logo from './logo.svg';
import { Router, Route, NavLink } from 'react-router-dom';
import { fetchApi } from './ApiCalls/apiCalls';
import './App.css';

class App extends Component {
  async componentDidMount() {
    const data = fetchApi();
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to Movie Tracker!!!!</h1>
        </header>
      </div>
    );
  }
}

export default App;
