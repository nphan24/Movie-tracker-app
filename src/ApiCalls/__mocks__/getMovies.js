import { cleanedMovie } from '../unit-test/mockData';

export const getMovies = jest.fn().mockImplementation(() => {
  return Promise.resolve({cleanedMovie});
});