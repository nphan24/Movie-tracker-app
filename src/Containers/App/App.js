import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Router, Route, NavLink } from 'react-router-dom';
import { getMovies } from '../../ApiCalls/getMovies';
import * as Actions from '../../Actions';
import CardContainer from '../CardContainer/CardContainer';
import { Login } from '../../Components/Login/Login';
import { addNewUser } from '../../ApiCalls/addNewUser';
import { signinUser } from '../../ApiCalls/signinUser';
import './App.css';

export class App extends Component {
  async componentDidMount() {
    const movies = await getMovies();
    this.props.retrieveMovies(movies);
  }

  render() {
    return (
      <main className="App">
        <header>
          <NavLink to='/login'>Login</NavLink>
          <Login />
        </header>
        <CardContainer />
      </main>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  retrieveMovies: movies => dispatch(Actions.postMovies(movies))
});

export default connect(null, mapDispatchToProps)(App);
