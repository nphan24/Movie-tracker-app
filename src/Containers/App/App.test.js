import React from 'react';
import { App, mapDispatchToProps } from './App';
import { shallow } from 'enzyme';
import { getMovies } from '../../ApiCalls/getMovies';
import { cleanedMovie } from '../../ApiCalls/unit-test/mockData';
import * as actions from '../../Actions';
jest.mock('../../ApiCalls/getMovies');


describe('App', () => {

  let wrapper;
  let mockRetrieveMovies;

  beforeEach(() => {
    mockRetrieveMovies = jest.fn();
    wrapper = shallow(
      <App retrieveMovies={mockRetrieveMovies}/>, 
      {disableLifecycleMethods: true});
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // it('should call getMovies on componentDidMount', () => {
  //   wrapper.instance().componentDidMount();
  //   expect(getMovies).toHaveBeenCalled();
  // });

  // it('should call retrieveMovies on componentDidMount', async () => {
  //   await wrapper.instance().componentDidMount();
  //   expect(mockRetrieveMovies).toHaveBeenCalled();
  // });
});

describe('mapDispatchToProps', () => {

  // it('should call dispatch with correct params', () => {
  //   const mockDispatch = jest.fn();
  //   const mockMovies = cleanedMovie;
  //   const expected = actions.postMovies(mockMovies);
  //   const mapped = mapDispatchToProps(mockDispatch);
  //   mapped.retrieveMovies(mockMovies);
  //   expect(mockDispatch).toHaveBeenCalledWith(expected);

  // });
});


