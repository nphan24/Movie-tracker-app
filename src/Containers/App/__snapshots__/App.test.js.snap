// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addFavorites={[Function]}
    favorites={
        Array [
            Object {
              "date": "Release Date: 2018-10-5",
              "image": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "movie_id": 10,
              "overview": "Summary: The tables are turned as four teenagers.",
              "title": "Fantastic Fours",
              "vote": "Rating: 2.5",
            },
            Object {
              "date": "Release Date: 2017-12-09",
              "image": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "movie_id": 101.350698,
              "overview": "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
              "title": "Jumanji: Welcome to the Jungle",
              "vote": "Rating: 6.5",
            },
          ]
    }
    history={
        Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
    }
    logout={[Function]}
    retrieveMovies={[Function]}
    signIn={[Function]}
    user={
        Object {
            "email": "r@r.r",
            "id": "5",
            "name": "r",
            "password": "r",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Route
        path="/"
        render={[Function]}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "path": "/",
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Route
          path="/"
          render={[Function]}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": false,
  },
}
`;

exports[`App should render correctly on url set to /favorites 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addFavorites={[Function]}
    favorites={
        Array [
            Object {
              "date": "Release Date: 2018-10-5",
              "image": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "movie_id": 10,
              "overview": "Summary: The tables are turned as four teenagers.",
              "title": "Fantastic Fours",
              "vote": "Rating: 2.5",
            },
            Object {
              "date": "Release Date: 2017-12-09",
              "image": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "movie_id": 101.350698,
              "overview": "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
              "title": "Jumanji: Welcome to the Jungle",
              "vote": "Rating: 6.5",
            },
          ]
    }
    history={
        Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 2,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "aa4v0i",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
    }
    location={
        Object {
            "hash": "",
            "pathname": "/favorites",
            "search": "",
            "state": undefined,
          }
    }
    logout={[Function]}
    retrieveMovies={[Function]}
    signIn={[Function]}
    user={
        Object {
            "email": "r@r.r",
            "id": "5",
            "name": "r",
            "password": "r",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Route
        path="/"
        render={[Function]}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "path": "/",
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Route
          path="/"
          render={[Function]}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`App should render correctly on url set to home / 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addFavorites={[Function]}
    favorites={
        Array [
            Object {
              "date": "Release Date: 2018-10-5",
              "image": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "movie_id": 10,
              "overview": "Summary: The tables are turned as four teenagers.",
              "title": "Fantastic Fours",
              "vote": "Rating: 2.5",
            },
            Object {
              "date": "Release Date: 2017-12-09",
              "image": "/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg",
              "movie_id": 101.350698,
              "overview": "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
              "title": "Jumanji: Welcome to the Jungle",
              "vote": "Rating: 6.5",
            },
          ]
    }
    history={
        Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 2,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "aa4v0i",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
    }
    location={
        Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
    }
    logout={[Function]}
    retrieveMovies={[Function]}
    signIn={[Function]}
    user={
        Object {
            "email": "r@r.r",
            "id": "5",
            "name": "r",
            "password": "r",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Route
        path="/"
        render={[Function]}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "path": "/",
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Route
          path="/"
          render={[Function]}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
